### PR TITLE
feat: add filesystem access permissions config for cheatcodes

### PR DESCRIPTION
**Pull request by mattsse**
## Motivation
configure if and what paths can be accessed via `write|readFile` related cheatcodes 

## Solution
* introduce `fs_permissions` config options that configures the type of access that is permitted (none, read-write, read, write)
* explicitly add paths that are allowed to access

by default fs access is disallowed and no paths are included in the allowed_paths set:

```toml
fs_permissions = { permission = false, allowed_paths = []}
```

this setting is explicitly opt-in

### CODE REVIEW

**Comment by mattsse** [comment]
forge-std currently fails due to `readFile` cheatcode usage

File: config/README.md [hunk 1]
```diff
@@ -153,6 +153,15 @@ sparse_mode = false
...
 root = "root"
+# Configres permissions for cheatcodes that touch the filesystem like `vm.writeFile`
+# `permission` restricts the type of access, possible values are
+#    `true` | "read-write" => `read` + `write` access allowed (`vm.readFile` + `vm.writeFile`)
+#    `false` | "none" => no access
+#    `read` => only read access (`vm.readFile`)
+#    `write` => only write access (`vm.writeFile`)
+# The `allowed_paths` further lists the paths that are considered, e.g. `./` represents the project root directory
+# By default _no_ fs access permission is granted, and _no_ paths are allowed
+fs_permissions = { permission = false, allowed_paths = [] }
```
Comment (Elyx0) [inline config/README.md:164]: how you considered per path control? so getting read-write wouldn't apply on all the paths?

Comment (Elyx0) [reply config/README.md:164]: `Configres` typo

Comment (mattsse) [reply config/README.md:164]: that makes sense, I thought of permission as a toggle for all paths but I guess it makes more sense to use this on a per path bases

Comment (mattsse) [reply config/README.md:164]: So I think I'll make `fs_permission` an array/table of `{ permission = "none|read|write|read-write", path = "" }`

**Comment by mattsse** [comment]
@Elyx0 revised it so that it can be configured per path:

```toml
fs_permissions = [{ access = "read", path = "./"}]
```

cc @gakonst

**Comment by gakonst** [comment]
Failing integration test think you need to override the test's config to the generated tempdir

```
Failing tests:
Encountered 5 failing tests in src/test/StdCheats.t.sol:StdCheatsTest
[FAIL. Reason: The path "/tmp/can_test_forge_std-0y9ayqm/lib/forge-std/out/StdCheats.t.sol/StdCheatsTest.json" is not allowed to be accessed for read operations.] testDeployCode() (gas: 3478)
[FAIL. Reason: Error != expected error: 'The path "/tmp/can_test_forge_std-0y9ayqm/lib/forge-std/out/StdCheats.t.sol/RevertingContract.json" is not allowed to be accessed for read operations.' != 'Test deployCode(string): Deployment failed.'] testDeployCodeFail() (gas: 5383)
[FAIL. Reason: The path "/tmp/can_test_forge_std-0y9ayqm/lib/forge-std/out/StdCheats.t.sol/StdCheatsTest.json" is not allowed to be accessed for read operations.] testDeployCodeNoArgs() (gas: 3461)
[FAIL. Reason: The path "/tmp/can_test_forge_std-0y9ayqm/lib/forge-std/out/StdCheats.t.sol/StdCheatsTest.json" is not allowed to be accessed for read operations.] testDeployCodeVal() (gas: 3480)
[FAIL. Reason: The path "/tmp/can_test_forge_std-0y9ayqm/lib/forge-std/out/StdCheats.t.sol/StdCheatsTest.json" is not allowed to be accessed for read operations.] testDeployCodeValNoArgs() (gas: 3465)
```

**Comment by mattsse** [comment]
tests currently fail because a new forge-std commit introduced additional warnings that fail the checks 

fixed here https://github.com/foundry-rs/forge-std/pull/167
should be resolved once merged

**Pull request merged**